### PR TITLE
[Keyboard] Add zfrontier/big_switch

### PR DESCRIPTION
**Pull request by carlxia**
## Description
A keyboard designed by zFrontier with RGB backlight and only one switch. Two keymaps are provided,
- keymap `default` has only the ENTER key.
- keymap `longtap` enables more features like switching RGB backlight mode and input a random slogan by different press modes.

## Types of Changes

- [x] Keyboard (addition or update)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/zfrontier/big_switch/config.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0A01
+#define DEVICE_VER 0x0005
+#define MANUFACTURER zfrontier
```
Comment (fauxpark) [inline keyboards/zfrontier/big_switch/config.h]:
```suggestion
#define MANUFACTURER zFrontier
```

File: keyboards/zfrontier/big_switch/config.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#define PRODUCT S01_Gaint_Key
```
Comment (fauxpark) [inline keyboards/zfrontier/big_switch/config.h]:
```suggestion
#define PRODUCT Red Comet Big Switch
```
maybe?

File: keyboards/zfrontier/big_switch/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+void keyboard_post_init_user(void) {
+    rgblight_enable_noeeprom();
+    rgblight_mode_noeeprom(RGBLIGHT_MODE_BREATHING);
+    rgblight_setrgb(128,0,0);
+}
+
+#ifdef TAP_DANCE_ENABLE
+qk_tap_dance_action_t tap_dance_actions[0];
+#endif
```
Comment (fauxpark) [inline keyboards/zfrontier/big_switch/keymaps/default/keymap.c]: I think if you add a rules.mk to this keymap containing just `TAP_DANCE_ENABLE = no` (or disable it at the keyboard level and enable it only for the `longtap` keymap), this block can be deleted.

File: keyboards/zfrontier/big_switch/keymaps/longtap/keymap.c [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            break;
+        case 6:
+            SEND_STRING("Go, Axis! With my unpleasant memories!");
+            break;
+        case 7:
+            SEND_STRING("Hello from zFrontier!");
+            break;
+    }
+
+    SEND_STRING(SS_TAP(X_ENTER));
```
Comment (fauxpark) [inline keyboards/zfrontier/big_switch/keymaps/longtap/keymap.c]:
```suggestion
    tap_code(KC_ENT);
```

File: keyboards/zfrontier/big_switch/rules.mk [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+# Otherwise, delete this section
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (fauxpark) [inline keyboards/zfrontier/big_switch/rules.mk]: Please delete this block.

File: keyboards/zfrontier/big_switch/keymaps/default/keymap.c (hunk 3 above)
Comment (carlxia) [reply keyboards/zfrontier/big_switch/keymaps/default/keymap.c]: Good suggestion!

**Comment by carlxia** [comment]
> Looking much better, thanks! Just a few small tweaks.

Thanks and Appreciate your detailed code review very much! @fauxpark 
I've added an overriding `rules.mk` in `keymap/longtap`.
Welcome if you have any new comments.

File: keyboards/zfrontier/big_switch/config.h [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* RGB light config */
+#define RGB_DI_PIN B6
+#define RGBLED_NUM 4
+
+#define RGBLIGHT_VAL_STEP 17
+
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/zfrontier/big_switch/config.h]: If you have this defined, then you don't need most of the rest of the defines, as this enables most of them.

File: keyboards/zfrontier/big_switch/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(KC_ENTER),
+};
+
+
+
+void keyboard_post_init_user(void) {
+    rgblight_enable_noeeprom();
+    rgblight_mode_noeeprom(RGBLIGHT_MODE_BREATHING);
+    rgblight_setrgb(128,0,0);
```
Comment (drashna) [inline keyboards/zfrontier/big_switch/keymaps/default/keymap.c]:
```suggestion
    rgblight_sethsv_noeeprom(0,0,128);
```

File: keyboards/zfrontier/big_switch/keymaps/longtap/keymap.c [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    LAYOUT( TD(TD_ENTER) ),
+};
+
+
+
+void keyboard_post_init_user(void) {
+    print("zFrontier S01.6 boot...\n");
+    rgblight_enable_noeeprom();
+    rgblight_mode_noeeprom(RGBLIGHT_MODE_BREATHING);
+    rgblight_setrgb(128,0,0);
```
Comment (drashna) [inline keyboards/zfrontier/big_switch/keymaps/longtap/keymap.c]:
```suggestion
    rgblight_sethsv_noeeprom(0,0,128);
```

File: keyboards/zfrontier/big_switch/config.h (hunk 6 above)
Comment (carlxia) [reply keyboards/zfrontier/big_switch/config.h]: Thanks and I've removed redundant things.

**Comment by carlxia** [comment]
@fauxpark Hi Ryan, can this PR be merged now? thanks.

**Comment by fauxpark** [comment]
@carlxia just that last thing on the default keymap, looks good to go otherwise.

File: keyboards/zfrontier/big_switch/keymaps/default/keymap.c (hunk 3 above)
Comment (carlxia) [reply keyboards/zfrontier/big_switch/keymaps/default/keymap.c]: removed

**Pull request merged**